R/checks.R

```diff
@@ -180,3 +180,54 @@ check_sparse_pmf_tail <- function(pmf, span = 5, tol = 1e-6) {
     )
   }
 }
+
+
+#' Check that supplied PMFs are not longer than the data
+#'
+#' @param ... Delay distributions
+#' @inheritParams estimate_infections
+#' @importFrom cli cli_warn col_red
+#'
+#' @returns Called for its side effects
+#' @keywords internal
+check_pmf_length_against_data <- function(..., data) {
+  delays <- list(...)
+  flat_delays <- do.call(c, delays)
+  # Track which component each delay came from
+  delay_names <- rep(names(delays), vapply(delays, EpiNow2:::ndist, numeric(1)))
+  names(flat_delays) <- delay_names
+  # Find the non-parametric distributions
+  np_delays <- which(unname(vapply(
+    flat_delays, function(x) {
+      get_distribution(x) == "nonparametric"
+    }, logical(1)
+  )))
+
+  if (length(np_delays) == 0) return(invisible())
+
+  # Check lengths and collect info about exceeding PMFs
+  pmf_longer_than_data <- vapply(flat_delays[np_delays], function(x) {
+    length(x$pmf) > nrow(data)
```

Comment:
don't we need to also check if the total reporting delay is longer than the data or the case where two distributions are passed to the generation time (not sure this is something you can actually do?)

Reply:
Good point. Fixed in f21ed951d88d872e17493ffcc4eaeba32a8910f9.